src/database/utils/threadDB.ts

```diff
@@ -40,34 +40,24 @@ export class ThreadDB {
       }
     }
   
-    threadReadUser = async(queryVariable: string, queryValue: string): Promise<User[]> => {
+    threadRead = async<T extends User | SafeData>(queryVariable: string, queryValue: string, collection: string): Promise<T[]> => {
       try{
-        const emailQuery = new Where(queryVariable).eq(queryValue);
-        const result: User[] = await this.connection.client.find(this.connection.threadId, 'Users', emailQuery);
+        let result: T[] = []
+        if(collection === 'Users' && queryValue !== ''){
+          const query = new Where(queryVariable).eq(queryValue);
+          result = await this.connection.client.find<T>(this.connection.threadId, 'Users', query);
```

Comment:
Use the collection variable instead of hard coding the collection name. Below as well.